frontend/src/components/Loader.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './Loader.css';
+
+function Loader({ loading }) {
+  return (
+    <>
+      {loading && (
+        <div role="status" className="overlay" aria-label="loading" />
```

Comment:
I agree this is a great usability improvement! 

It'd be cool if this `aria-label` could be customized so that screen readers could announce _what_ is loading. Maybe also throw `aria-live="polite"` in there?